rfc/helm-import.md

```diff
@@ -0,0 +1,515 @@
+    D --yes--> F(get chart details from url)
+    F --> G(create temporary directory)
+    G --> H(helm pull chart \n in tmp dir)
+    H --> I{is \n values.schema.json \n present}
+    I --no--> J(schema path is set \n to undefined)
+    I --yes--> K(schema path is set)
+    L(import continues)
+    J --> L
+    K --> L
+    L --> M(generate typescript)
+    M --> N(emit 'values' struct)
+    N --> O(emit construct props)
+    O --> P(emit construct)
+    P --> Q(render to file and save)
+    Q --> R{is language typescript}
+    R --yes--> S(import finishes)
+    R --no--> T(transpile to target language)
+    T --> S
+    C --> S
+```
+
+### Is this a breaking change?
+
+This is not a breaking change. This is adding new functionality to the cdk8s cli import command. 
+
+### What alternative solutions did you consider?
+
+* **Using omit utility type**
+
+  In the proposed solution we are recreating a similar interface to [HelmProps](https://github.com/cdk8s-team/cdk8s-core/blob/2.x/src/helm.ts#L15-L77). This is unideal since it adds redundancy. The reason for doing this is because:
+  
+    1. We already would know the `chart` value and it would not make sense to ask the user for it which is required in HelmProps.
+    2. `values` could have data types associated with its properties which is not supported in HelmProps.
+  
+  I initially used [Omit utility type](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) and it works but just for typescript. When trying to transpile it to other languages it gives out an error due to jsii limitation. 
+
+* **--helm as a flag**
+
+  Instead of using `cdk8s import helm:url`, I initially thought of using a flag `--helm` where if present, it would mean that the url is supposed to be leading to a helm chart. This adds a flag to the import command and keeping it as `helm:` makes it similar to what we have for `github:`.
+
+* **Reusing ApiObjectDefinition**
+
+  In the proposed solution, I have added `HelmObjectDefinition` as an interface instead of reusing and adding to [ApiObjectDefinition](https://github.com/cdk8s-team/cdk8s-cli/blob/2.x/src/import/codegen.ts#L10-L37). I think the two interface represent different things and it would not make sense to combine them for this feature.
+
+### What are the drawbacks of this solution?
+
+* **Interface similar to HelmProps**
+  
+  As mentioned in the previous sections, there is an interface similar to HelmProps in the proposed solution due to a jsii limitation. We can use `mgrain's` [project](https://github.com/mrgrain/jsii-struct-builder) to reduce code but the emitted interface would still be similar to HelmProps. 
+
+* **Multiple values.schema.json**
+
+  We are considering only the `values.schema.json` present at the root of the chart since helm by default looks for `values.yaml` at the root of the chart. I have seen multiple values.yaml files in a chart's sub-folders and these can have schema associated with them. 
```

Comment:
Can you give an example?

Reply:
I was not able to find the chart that I encountered this in, will keep looking. 

But, I found a [GH issue](https://github.com/helm/helm/issues/11815) for requesting support for multiple schema files and that was not entertained. So I probably misunderstood.

Reply:
I think you might have seen sub-charts. However, this does bring up the question of what to do with sub charts? 

If a chart has sub-charts, we need to make sure those sub-charts can be configured as well. And if the sub-chart has a `values.schema.json` - that configuration should be typed as well. Lets talk about this online.